main.py

```diff
@@ -25,6 +25,30 @@ def __contains__(self, item):
     def __iter__(self):
         yield self
 
+def validate_stability(Kp, Ki):
+    if (((Kp + Ki)*(Kp + Ki)) < (4*Ki)):
+        return True
+    else:
+        return False
+
+def draw_stable_kp_ki():
+    stable = False
+    while not stable:
+        Kp = random.uniform(0, config.gen_max_kp_stable)
+        Ki = random.uniform(0, config.gen_max_ki_stable)
+        if validate_stability(Kp, Ki):
+            stable = True
+    return Kp, Ki
+
+def redefine_kp_ki_to_stable(Kp, Ki):
+    stable = False
+    while not stable:
+        Kp = Kp - config.reduction_determinant * Kp
+        Ki = Ki - config.reduction_determinant * Ki
+        if validate_stability(Kp, Ki):
+            stable = True
+    return Kp, Ki
```

Comment:
Should I move 28-50 to a separate pyton file and include? Just 3 functions/25 lines, separate functionality on the other hand.

Reply:
You can - but maybe it'd be easier to create a function that will generate stable pair of Kp and Ki as an API for this file?